README.md

```diff
@@ -80,7 +80,58 @@ Alternatively, you can specify `html.root` in your configuration and this will b
 
 You can add or override the default [webpack-isomorphic-tools](https://github.com/halt-hammerzeit/webpack-isomorphic-tools) configuration, by providing a `toolsConfigPath` value to your `config.js`.
 
-#### Scripts
+### Plugins
+
+Hooks can be used in order to override the default implementation of certain functions. Files that define hooks are called plugins and must be listed in the `plugins` property of `config/universal-redux.config.js`. The plugin paths are resolved as follow:
+
+ - ${project-src}/{path}
+ - ${project-root}/node_modules/universal-redux-plugin-{path}
+ - ${project-root}/node_modules/{path}
+
+The plugin file should register the implementations of the functions it wants to override as follow:
+
+```js
+import { hooks, register } from 'universal-redux/lib/hooks';
+
+register(hooks.client.GENERATE_ROOT_COMPONENT, () => {
+  //Generate the component
+  return { mycomponent };
```

Comment:
It should be a react element so it can be either 
```js
const mycomponent = <MyComponent />
return { root: mycomponent };
``` 
or 
```js
return { root: <MyComponent store={store} routes={routes} /> }
```
I definitely need to fix that.